ui/spells/spells.scss

```diff
@@ -130,110 +132,50 @@
     margin-top: -1px;
   }
 
-  $grid-padding: 2px;
-  $grid-column-gap: 2px;
-
-  @mixin spells-header($grid-template) {
-    display: grid;
-    grid-template-columns: $grid-template;
-    column-gap: $grid-column-gap;
-    background: $primary-highlight-background-color;
-    color: $primary-highlight-color;
-    padding: $grid-padding;
-
-    > div {
-      text-align: center;
-      margin: auto;
-      font-weight: bold;
-      width: 100%;
-    }
-  }
-
-  @mixin spells-rows($grid-template, $column-count) {
-    display: grid;
-    grid-template-columns: $grid-template;
-    row-gap: 0;
-    column-gap: $grid-column-gap;
-    background: transparent;
-    input, input[type=number], textarea, select {
-      width: 100%;
-      margin-left: unset;
-      height: auto;
-      background-color: $primary-background-color;
-    }
-    input[type=text] {
-      text-align: left;
-    }
-    input, span, select {
-      white-space: nowrap;
-    }
-    span {
-      text-align: center;
-      margin: auto 0;
-    }
-    button, button[type=roll] {
-      margin: 0;
-    }
-    .spell-row.conditional-one {
-      // This makes the notes span all columns
-      grid-column-start: 1;
-      grid-column-end: ($column-count + 1);
-    }
-  }
-
-  @mixin spells-grid($spell-type, $grid-template, $column-count) {
-    .spells-header-grid.#{$spell-type} {
-      @include spells-header($grid-template);
-    }
-
-    div[data-groupname="repeating_#{$spell-type}"] > div {
-      @include spells-rows($grid-template, $column-count);
-    }
-  }
-
-  $arcane-grid-template:
+  $arcane-grid-template: getResponsiveGridWidths(
     // Discipline?
-    minmax(0, 1fr)
+    1fr,
     // Name
-    minmax(0, 4fr)
+    4fr,
     // Ability
-    minmax(0, 0.75fr)
+    0.75fr,
     // Mod
-    minmax(0, 1fr)
+    1fr,
     // (Dice)
-    minmax(0, 0.5fr)
+    0.5fr,
     // EP
-    minmax(0, 1fr)
+    1fr,
     // Cast Time
-    minmax(0, 1fr)
+    1fr,
     // Touch/ Ray/ Missile
-    minmax(0, 2fr)
+    2fr,
     // Range
-    minmax(0, 1fr)
+    1fr,
     // Target
-    minmax(0, 1fr)
+    1fr,
     // Damage/ Duration
-    minmax(0, 1.5fr)
+    1.5fr,
     // N(otes)
-    minmax(0, 0.5fr)
+    0.5fr,
     // TP
-    minmax(0, 0.5fr)
+    0.5fr,
     // CP
-    minmax(0, 0.5fr)
+    0.5fr,
     // Exper.
-    minmax(0, 1fr)
+    1fr,
     // IQ ±
-    minmax(0, 0.5fr)
+    0.5fr,
     // Base
-    minmax(0, 1fr);
+    1fr,
+  );
 
-  @include spells-grid('arcane', $arcane-grid-template, 18);
+  @include epic-table-grid('arcane', $arcane-grid-template);
```

Comment:
I eliminated the need to specify the length of the grid template array. This 18 was actually incorrect. It should have been 17.
It caused a misalignment in the UI:
| Old | new |
| ![image](https://user-images.githubusercontent.com/5629800/209588625-167c78f4-e7c7-444f-8ff4-e23f98e8c4e6.png) | ![image](https://user-images.githubusercontent.com/5629800/209588643-3eeb943d-293f-4f28-8e7d-e97737bb4402.png) |